File: LeetCode/Mathematics/LC-1342 Steps to Reduce Number to Zero/LC1342.java
class LC1342 {
    public static int numberOfSteps(int num) {
        int steps = 0;
        while(num>0){
            steps++;
            if(num%2 == 1)
                num--;
            else
                num /= 2;
        }
        return steps;
    }

    public static void main(String[] args){
        System.out.println(numberOfSteps(14));
        System.out.println(numberOfSteps(8));
        System.out.println(numberOfSteps(123));
    }
}

File: LeetCode/Mathematics/LC-1342 Steps to Reduce Number to Zero/LC1342.cs
using System;

class LC1342
{
    public static int NumberOfSteps(int num)
    {
        int steps = 0;
        while (num > 0)
        {
            steps++;
            if (num % 2 == 1)
                num--;
            else
                num /= 2;
        }
        return steps;
    }

    public static void Main(string[] args)
    {
        Console.WriteLine(NumberOfSteps(14));   // Output: 6
        Console.WriteLine(NumberOfSteps(8));    // Output: 4
        Console.WriteLine(NumberOfSteps(123));  // Output: 12
    }
}
